src/signal.rs

```diff
@@ -293,3 +295,67 @@ mod spin_button {
         f(&from_glib_borrow(this))
     }
 }
+
+pub trait OverlaySignals {
+    fn connect_get_child_position<F>(&self, f: F) -> SignalHandlerId
+    where
+        F: Fn(&Self, &Widget) -> Option<Rectangle> + 'static;
+}
+
+mod overlay {
+    use gdk::Rectangle;
+    use ffi::{GtkOverlay, GtkWidget};
+    use gdk_ffi::GdkRectangle;
+    use glib::signal::{connect, SignalHandlerId};
+    use glib::translate::*;
+    use glib::object::Downcast;
+    use glib_ffi::{GFALSE, GTRUE};
+    use std::mem::transmute;
+    use std::ptr;
+    use glib_ffi::{gboolean, gpointer};
+    use IsA;
+    use Object;
+    use Overlay;
+    use Widget;
+
+    impl<O: IsA<Overlay> + IsA<Object>> ::OverlaySignals for O {
+        fn connect_get_child_position<F>(&self, f: F) -> SignalHandlerId
+        where
+            F: Fn(&Self, &Widget) -> Option<Rectangle> + 'static,
+        {
+            unsafe {
+                let f: Box<Box<Fn(&Self, &Widget) -> Option<Rectangle> + 'static>> =
+                    Box::new(Box::new(f));
+                connect(
+                    self.to_glib_none().0,
+                    "get-child-position",
+                    transmute(get_child_position_trampoline::<Self> as usize),
+                    Box::into_raw(f) as *mut _,
+                )
+            }
+        }
+    }
+
+    unsafe extern "C" fn get_child_position_trampoline<T>(
+        this: *mut GtkOverlay,
+        widget: *mut GtkWidget,
+        allocation: *mut GdkRectangle,
+        f: gpointer,
+    ) -> gboolean
+    where
+        T: IsA<Overlay> + IsA<Object>,
+    {
+        callback_guard!();
+        let f: &&(Fn(&T, &Widget) -> Option<Rectangle> + 'static) = transmute(f);
+        match f(
+            &Overlay::from_glib_borrow(this).downcast_unchecked(),
+            &from_glib_borrow(widget),
+        ) {
+            Some(rect) => {
+                ptr::write(allocation, transmute(rect));
```

Comment:
Maybe `write_unaligned` need used instead?

Reply:
No, the rectangle destination is required to be properly aligned. It's otherwise a bug in the code that emitted the signal.

But why is the transmute necessary? Don't we have the normal translation traits implemented for Rectangle? Otherwise looks all good

Reply:
`*allocation = *rect.to_glib_none().0` will fails with `error[E0507]: cannot move out of dereference of raw pointer`,
using dereference instead `transmute` will produce same error.
Currently no direct conversion to `GtkRectangle` other that `to_glib_none`.

Reply:
`ptr::write(allocation, rect.to_glib_none().0)`?

Reply:
Thanks, I missed that possibility.